syntax = "proto3";

package kv.proto;

service Settings {
  rpc set (set_request) returns (set_response) {}
  rpc get (get_request) returns (get_response) {}
}

message data {
	string key = 1;
	string value = 2;
};

message set_request {
	data req = 1;
};

message set_response {
	enum error_t {
		ok = 0;
		internal = 1;
	};
	error_t error = 1;
};

message get_request {
	string key = 1;
};

message get_response {
	enum error_t {
		ok = 0;
		not_found = 1;
		internal = 2;
	};
	error_t error = 1;
	data req = 2;
};

message req_envelope {
	enum msg_type {
		set_request_t = 0;
		set_response_t = 1;
		get_request_t = 2;
		get_response_t = 3;
	};
	msg_type type = 1;
	set_request set_req = 2;
	set_response set_resp = 3;
	get_request get_req = 4;
	get_response get_resp = 5;
};
